File: app/java/src/main/java/com/example/production/domain/model/plan/AllocationType.java
package com.example.production.domain.model.plan;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum AllocationType {
    INVENTORY("在庫"),
    PURCHASE_ORDER("発注残"),
    MANUFACTURING_ORDER("製造残");

    private final String displayName;

    public static AllocationType fromDisplayName(String displayName) {
        for (AllocationType type : values()) {
            if (type.displayName.equals(displayName)) {
                return type;
            }
        }
        throw new IllegalArgumentException("Unknown allocation type: " + displayName);
    }
}

File: app/csharp/src/ProductionManagement.Domain/Models/Plan/AllocationType.cs
namespace ProductionManagement.Domain.Models.Plan;

/// <summary>
/// 引当区分
/// </summary>
public enum AllocationType
{
    /// <summary>在庫</summary>
    Inventory,
    /// <summary>発注残</summary>
    PurchaseOrder,
    /// <summary>製造残</summary>
    ManufacturingOrder
}

/// <summary>
/// AllocationType 拡張メソッド
/// </summary>
public static class AllocationTypeExtensions
{
    private static readonly Dictionary<AllocationType, string> DisplayNames = new()
    {
        { AllocationType.Inventory, "在庫" },
        { AllocationType.PurchaseOrder, "発注残" },
        { AllocationType.ManufacturingOrder, "製造残" }
    };

    private static readonly Dictionary<string, AllocationType> FromDisplayNames =
        DisplayNames.ToDictionary(x => x.Value, x => x.Key);

    public static string GetDisplayName(this AllocationType type) => DisplayNames[type];

    public static AllocationType FromDisplayName(string displayName)
    {
        if (FromDisplayNames.TryGetValue(displayName, out var type))
        {
            return type;
        }
        throw new ArgumentException($"Unknown allocation type: {displayName}");
    }
}
